forte/data/readers/tests/race_multi_choice_qa_reader_test.py

```diff
@@ -0,0 +1,55 @@
+import json
+import unittest
+from typing import Iterator
+
+from forte.data.readers import RACEMultiChoiceQAReader
+from forte.data.data_pack import DataPack
+from ft.onto.race_mutli_choice_qa_ontology import Article, Question
+
+
+class RACEMultiChoiceQAReaderTest(unittest.TestCase):
+    def setUp(self):
+        self.dataset_path = \
+            "forte/data/readers/tests/data_samples/race_multi_choice_qa"
+
+    def test_reader_no_replace_test(self):
+        # Read with no replacements
+        reader = RACEMultiChoiceQAReader()
+        data_packs: Iterator[DataPack] = reader.iter(self.dataset_path)
+        file_paths: Iterator[str] = reader._collect(self.dataset_path)
+
```

Comment:
Interesting. But we normally test the components by running them as part of the pipeline (e.g. https://github.com/asyml/forte/blob/8e75b567f805e201aa5551d7268c9572ff5a47ee/forte/processors/tests/bert_based_query_creator_test.py). That way we can have a better test on whether it is integrated into the pipeline correctly.

But I do think a real unit test like this is also OK.

Reply:
Yeah I was a bit confused about that because some of the reader tests are using the Pipeline while others are just checking the functionality of that particular reader.

I had discussed this with @mgupta1410 and we had agreed that since these are just the reader unit tests they should only be checking the functionality of the reader. I can change the test though if you guys think we should have the pipeline tests.

Reply:
OK let's do it this way and we can discuss a consistent way in the future.